lua/ttt2/libraries/entspawnscript.lua

```diff
@@ -0,0 +1,1072 @@
+		-- update amount of spawns on clients
+		entspawnscript.UpdateSpawnCountOnClients()
+	end
+
+	if not shouldSync then return end
+
+	net.Start("ttt2_add_spawn_ent")
+	net.WriteUInt(spawnType, 4)
+	net.WriteUInt(entType, 4)
+	net.WriteVector(pos)
+	net.WriteAngle(ang)
+	net.WriteUInt(ammo, 8)
+
+	if SERVER then
+		net.Send(entspawnscript.GetReceivingPlayers(ply))
+	else -- CLIENT
+		net.SendToServer()
+	end
+end
+
+---
+-- Updates an existing spawn point in the spawn point table. Does nothing if the
+-- spawn point does not exist. If enabled, it is automatically synced to the server / client.
+-- @note Changes are not automatically written to disk. This is done once the spawn edit is done. Use
+-- {entspawnscript.UpdateSpawnFile} to trigger it manually.
+-- @param number spawnType The spawn type of the spawn that should be updated
+-- @param number entType The entity type of the spawn that should be updated
+-- @param number id The numeric id of the spawn that should be removed
+-- @param Vector pos The position vector of the spawn that should be updated
+-- @param Angle ang The angle of the spawn that should be updated
+-- @param[default=0] number ammo The amount of ammo of the spawn that should be updated,
+-- only relevant to weapon spawns
+-- @param boolean shouldSync Set to true if it should be synced, set to false if it shouldn't
+-- @param Player ply The player that attempts to add the spawn. Only relevant if synced from
+-- the server to the client for net performance reasons
+-- @realm shared
+function entspawnscript.UpdateSpawn(spawnType, entType, id, pos, ang, ammo, shouldSync, ply)
+	if not spawnEntList or not spawnEntList[spawnType] or not spawnEntList[spawnType][entType] then return end
+
+	local listEntry = spawnEntList[spawnType][entType]
+
+	if not listEntry[id] then return end
+
+	pos = pos or listEntry[id].pos
+	ang = ang or listEntry[id].ang
+	ammo = ammo or listEntry[id].ammo
+
+	listEntry[id] = {
+		pos = pos,
+		ang = ang,
+		ammo = ammo
+	}
+
+	if not shouldSync then return end
+
+	net.Start("ttt2_update_spawn_ent")
+	net.WriteUInt(spawnType, 4)
+	net.WriteUInt(entType, 4)
+	net.WriteUInt(id, 32)
+	net.WriteVector(pos)
+	net.WriteAngle(ang)
+	net.WriteUInt(ammo, 8)
+
+	if SERVER then
+		net.Send(entspawnscript.GetReceivingPlayers(ply))
+	else -- CLIENT
+		net.SendToServer()
+	end
+end
+
+---
+-- Deletes all spawn points on the map. Automatically syncs between client and server.
+-- @note Deleting all spawns does update the weapon spawn file immediately. Be careful.
+-- @realm shared
+function entspawnscript.DeleteAllSpawns()
+	if CLIENT then
+		net.Start("ttt2_delete_all_spawns")
+		net.SendToServer()
+	else
+		entspawnscript.SetSpawns({
+			[SPAWN_TYPE_WEAPON] = {},
+			[SPAWN_TYPE_AMMO] = {},
+			[SPAWN_TYPE_PLAYER] = {}
+		})
+
+		entspawnscript.WriteFile(spawnEntList, settingsList, spawndir .. gameGetMap() .. ".txt")
+
+		net.SendStream("TTT2_WeaponSpawnEntities", spawnEntList, entspawnscript.editingPlayers)
```

Comment:
This could be synced with ttt2net as a table so it is automatically sent, when it updates.